homeassistant/components/transmission.py

```diff
@@ -0,0 +1,189 @@
+"""
+Component for monitoring the Transmission BitTorrent client API.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/transmission/
+"""
+from datetime import timedelta
+
+import logging
+import voluptuous as vol
+
+from homeassistant.util import Throttle
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import discovery
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_NAME,
+    CONF_PORT,
+    CONF_USERNAME,
+    CONF_MONITORED_CONDITIONS,
+)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['transmissionrpc==0.11']
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'transmission'
+DATA_TRANSMISSION = 'TRANSMISSION'
+
+DEFAULT_NAME = 'Transmission'
+DEFAULT_PORT = 9091
+TURTLE_MODE = 'turtle_mode'
+
+SENSOR_TYPES = {
+    'active_torrents': ['Active Torrents', None],
+    'current_status': ['Status', None],
+    'download_speed': ['Down Speed', 'MB/s'],
+    'paused_torrents': ['Paused Torrents', None],
+    'total_torrents': ['Total Torrents', None],
+    'upload_speed': ['Up Speed', 'MB/s'],
+    'completed_torrents': ['Completed Torrents', None],
+    'started_torrents': ['Started Torrents', None],
+}
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
```

Comment:
Username and password would still be optional as it's not a requirement of the Transmission client to set them.